docs/book/v2/routing-middleware.md

```diff
@@ -0,0 +1,263 @@
+                'options' => [
+                    'route'    => '/closure',
+                    'defaults' => [
+                        'controller' => PipeSpec::class,
+                        'middleware' => static function (
+                            ServerRequestInterface $request,
+                            RequestHandlerInterface $handler
+                        ): ResponseInterface {
+                            return $handler->handle($request);
+                        },
+                    ],
+                ],
+            ],
+        ],
+    ],
+];
+```
+
+## Mixed `PipeSpec` Arguments
+
+In this example multiple different possible parameters are passed to `PipeSpec`:
+
+```php
+use Application\Handler\AlbumDetailHandler;
+use Laminas\Mvc\Middleware\PipeSpec;
+use Laminas\Router\Http\Literal;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+
+return [
+    'router' => [
+        'routes' => [
+            'instance-pipespec-dev-env' => [
+                'type' => Literal::class,
+                'options' => [
+                    'route' => '/instance-pipespec-dev-env',
+                    'defaults' => [
+                        'controller' => PipeSpec::class,
+                        'middleware' => new PipeSpec(
+
+                            // id/alias of service in container
+                            'container_id_for_album_middleware',
+
+                            // anonymous class implementing a MiddlewareInterface
+                            new class implements MiddlewareInterface {
+                                public function process(
+                                    ServerRequestInterface $request,
+                                    RequestHandlerInterface $handler
+                                ): ResponseInterface {
+                                    return $handler->handle($request);
+                                }
+                            },
+
+                            // Closure
+                            static function (
+                                ServerRequestInterface $request,
+                                RequestHandlerInterface $handler
+                            ): ResponseInterface {
+                                return $handler->handle($request);
+                            },
+
+                            // class-string of an Handler which will be resolved by the container
+                            AlbumDetailHandler::class,
+                        ),
+                    ],
+                ],
+            ],
+        ],
+    ],
+];
+```
+
+Please note that if you pass instances of middleware or handlers directly, it is not compatible with config caching. You
+should use that for development only. The recommended variant is passing container managed middleware
+like `SomeRequestHandler::class`.
```

Comment:
Make this into a note, please, so it gets called out more prominently. This is generally done by doing it in a blockquote, with a heading; I don't know if @froschdesign has other markup you can specify to call it out.